lib/mongo/collection.rb

```diff
@@ -537,8 +537,15 @@ def inspect
     # @since 2.0.0
     def insert_one(document, opts = {})
       client.send(:with_session, opts) do |session|
-        write_concern = write_concern_with_session(session)
+
+        if opts.has_key?(:write_concern) ? (write_concern = WriteConcern.get(opts[:write_concern])) : (write_concern = write_concern_with_session(session))
```

Comment:
We prefer `key?` over `has_key?` and spelling out the if-then-else over writing ternaries.